plugins/pangea/textGuard.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  HookEventType,
+  PluginContext,
+  PluginHandler,
+  PluginParameters,
+} from '../types';
+import { post, getText } from '../utils';
+import { VERSION } from './version';
+
+export const handler: PluginHandler = async (
+  context: PluginContext,
+  parameters: PluginParameters,
+  eventType: HookEventType,
+  options: {}
+) => {
+  let error = null;
+  let verdict = false;
+  let data = null;
+  try {
+    if (!parameters.credentials?.domain) {
+      throw Error(`'parameters.credentials.domain' must be set`);
+    }
+
+    if (!parameters.credentials?.apiKey) {
+      throw Error(`'parameters.credentials.apiKey' must be set`);
+    }
+
+    // TODO: Update to v1 once released
+    const url = `https://ai-guard.${parameters.credentials.domain}/v1beta/text/guard`;
+
+    const text = getText(context, eventType);
+    if (!text) {
+      throw Error(`request or response text is empty`);
```

Comment:
this shouldn't block the whole request (i.e throwing error may close the request), instead you just return the plugin result with error field.

```
return {error: "request or response text is empty", verdict: true, data}
```